st2-ask/notification_rule.yaml

```diff
@@ -0,0 +1,21 @@
+---
+name: "notify_slack_execution_change"
+pack: "st2"
+description: "Watches for an execution status of "awaits" for a certain action ref
+enabled: true
+
+trigger:
+  type: core.st2.generic.actiontrigger
+
+criteria:
+  trigger.action_ref:
+      type: equals
+      pattern : st2.ask
+  trigger.status:
+      type: equals
+      pattern : awaits
+
+action:
+  ref: slack.post_message
+  parameters:
+    message: '{{trigger.execution_id}} is awaiting an approval action'
```

Comment:
> So, in StackStorm/discussions#236 the idea of creating a new runner where this could be done was discussed, and it seemed like the consensus was exposing the user to a new runner was equally bad. So with no new runner, and without executing notification actions from the st2 action, to me, this moves notification into the realm of the stackstorm user, meaning rules. Let me modify the existing examples to illustrate how this will be done with rules under the assumption that this is the way forward for now - then we can step back and make sure the overall approach is the right one

So even for rules, how will you target a specific st2.ask execution to notify a specific user?  That's the question.  So in the example w/ Bob and Fred, how will the rule notify Bob to respond to st2.ask execution ABC and Fred to respond to st2.ask execution XYZ.